components/dashboard/src/projects/NewProject.tsx

```diff
@@ -428,18 +462,54 @@ export default function NewProject() {
         </div>);
     }
 
-    return (<div className="flex flex-col w-96 mt-24 mx-auto items-center">
-        <h1>New Project</h1>
+    const onNewWorkspace = async () => {
+        const redirectToNewWorkspace = () => {
+            // instead of `history.push` we want forcibly to redirect here in order to avoid a following redirect from `/` -> `/projects` (cf. App.tsx)
+            const url = new URL(window.location.toString());
+            url.pathname = "/";
+            url.hash = project?.cloneUrl!;
+            window.location.href = url.toString();
+        }
+        redirectToNewWorkspace();
+    }
+
+    if (!project) {
+        return (<div className="flex flex-col w-96 mt-24 mx-auto items-center">
 
-        {!selectedRepo && renderSelectRepository()}
+            <>
+                <h1>New Project</h1>
 
-        {selectedRepo && !selectedTeamOrUser && renderSelectTeam()}
+                {!selectedRepo && renderSelectRepository()}
 
-        {selectedRepo && selectedTeamOrUser && (<div></div>)}
+                {selectedRepo && !selectedTeamOrUser && renderSelectTeam()}
+
+                {selectedRepo && selectedTeamOrUser && (<div></div>)}
+            </>
+
+            {isBitbucket() && renderBitbucketWarning()}
+
+        </div>);
+    } else {
+        const projectLink = User.is(selectedTeamOrUser) ? `/projects/${project.slug}` : `/t/${selectedTeamOrUser?.slug}/${project.slug}`;
+        const location = User.is(selectedTeamOrUser) ? "" : (<> in <a className="gp-link" href={`/t/${selectedTeamOrUser?.slug}/projects`}>{selectedTeamOrUser?.name}</a></>);
```

Comment:
(Suggestion to go with the other suggestion:)

```suggestion
        const location = User.is(selectedTeamOrUser) ? "" : (<> in team <a className="gp-link" href={`/t/${selectedTeamOrUser?.slug}/projects`}>{selectedTeamOrUser?.name}</a></>);
```

Reply:
sounds good!